src/table/body-cell/index.tsx

```diff
@@ -56,6 +56,28 @@ function TableCellEditable<ItemType>({
   const [hasFocus, setHasFocus] = useState(false);
   const showIcon = hasHover || hasFocus;
 
+  const prevSuccessfulEdit = useRef(successfulEdit);
+  const prevHasFocus = useRef(hasFocus);
```

Comment:
We have a hook for this: https://github.com/cloudscape-design/components/blob/main/src/internal/hooks/use-previous/index.ts

Reply:
I wasn't aware of the `usePrevious`, thanks for the shout-out - updating the PR.